pkg/text/text.go

```diff
@@ -0,0 +1,178 @@
+// Package text is a set of utility functions for text processing and outputting to the terminal.
+package text
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/muesli/reflow/ansi"
+	"github.com/muesli/reflow/truncate"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const (
+	ellipsis            = "..."
+	minWidthForEllipsis = len(ellipsis) + 2
+)
+
+var indentRE = regexp.MustCompile(`(?m)^`)
+var whitespaceRE = regexp.MustCompile(`\s+`)
+
+// Indent returns a copy of the string s with indent prefixed to it, will indent each line
+// in the string.
+func Indent(s, indent string) string {
```

Comment:
Open to better naming here. Additionally should we remove any currently prefixed whitespace from each line before adding the indent? That would make it so there is consistent indenting for each line of the string. This is not always what users would want, but perhaps it is the most common use case?